Feat: Разум для обезьян

PhantornRU opened the pull request:
Авоут пройден и одобрен:
https://discord.com/channels/617003227182792704/755125334097133628/1014916202784620694

Также пройден Авоут по наименованию питомцев, встроен сразу сюда, т.к. связан также с обезьянами:
https://discord.com/channels/617003227182792704/755125334097133628/1017403722294362223

## What Does This PR Do
- Зелье разума дает разум обезьянам и прицепляет их к хозяину
- Обезьяны ограничены в умственных способностях, когда они становятся разумными через зелья.
- Низшие РАЗУМНЫЕ формы получили возможность понимать человеческую речь
- Зелья разума на животных, теперь вносят памятки в их имена кому они служат
- Становление петов теперь логируется, кто их сделал.
- Превращение магом в мартышку - позволяет помнить прошлую жизнь.
- Небольшой рефактор и вынос макак и низших форм в отдельный файл.
- Ауры, огонь и скелеты подогнаны под обезьян

## Why It's Good For The Game
- Новые возможности для РП и путей выполнения задач со стороны Антагов. Но они по прежнему не могут лазать по вентиляциям с предметами.
- Ксенобиология теперь может делать не только рабов-големов, но и мартых.
- Мартыхи имеют хозяина и не могут по своей воле дебоширить.
- Ape. Together. Strong.
- Можно устраивать обезьяньи нелегальные подпольные бои.

## Images of changes
![image](https://user-images.githubusercontent.com/41479614/187676760-5a01bf9e-9a50-43a5-a93b-30a900fd4f78.png)
![image](https://user-images.githubusercontent.com/41479614/188380984-36e209f8-d38c-47d2-978e-66de066075f7.png)

PhantornRU committed:
Feat: Разум для обезьян

- Зелье разума дает разум обезьянам и прицепляет их к хозяину
- Обезьяны ограничены в умственных способностях, когда они становятся разумными через зелья.
- Низшие РАЗУМНЫЕ формы получили возможность понимать человеческую речь
- Зелья разума на животных, теперь вносят памятки в их имена кому они служат
- Становление петов теперь логируется, кто их сделал.
- Превращение магом в мартышку - позволяет помнить прошлую жизнь.
- Небольшой рефактор и вынос макак и низших форм в отдельный файл.

PhantornRU committed:
Фикс изменения строки раньше времени.

larentoun commented on code/modules/mob/living/carbon/human/species/monkey.dm:
```diff
@@ -8,7 +8,7 @@
 	damage_mask = 'icons/mob/human_races/masks/dam_mask_monkey.dmi'
 	blood_mask = 'icons/mob/human_races/masks/blood_monkey.dmi'
-	language = null
+	language = "Galactic Common"	//бореры теперь "не забывают" всегалактический, а разумные мартышки понимают что им говорят. Они по прежнему общаются на обезьянем/фарванском
```
POG

larentoun commented on code/modules/research/xenobiology/xenobiology.dm:
```diff
@@ -183,45 +183,83 @@
-	to_chat(user, "<span class='notice'>You offer [src] sentience potion to [SM]...</span>")
-	being_used = 1
+	if (isanimal(M))
+		var/mob/living/simple_animal/SM = M
```
Это SuperMonkey, или SimpleMob? КЫК

larentoun commented on code/modules/research/xenobiology/xenobiology.dm:
```diff
@@ -183,45 +183,83 @@
+		if(!src)
+			return
+
+		if(candidates.len)
```
Ну ПУПС, есть же невероятное length()

PhantornRU replied:
Оставил оригинальный код pepeChill, чтобы не было разнобойности

PhantornRU replied:
Это проверка на "не нулевость". len для этого и нужна, она возвращает значение FALSE при NULL.
length() в данном случае будет делать тоже самое, только больше буковок будет написано. Оба эти значения возвращают нули, если списки будут пусты. Если нет, то продолжают работу.
поэтому шо так, шо так. Но в нашем случае в ДМе принято использоваться именно что .len
Да и мне так удобнее сразу у объекты вызывать нужные внутренние функции, а не притягивать сторонние функции.

larentoun replied:
Валлат (как и ТГшники) переехали на length(). Считается новым стандартом

larentoun commented on code/modules/research/xenobiology/xenobiology.dm:
```diff
@@ -183,45 +183,83 @@
-	to_chat(user, "<span class='notice'>You offer [src] sentience potion to [SM]...</span>")
-	being_used = 1
+	if (isanimal(M))
+		var/mob/living/simple_animal/SM = M
```
А все-таки ревью оставлю, что за SM то? SuperMonkey? SimpleMob не подходит

PhantornRU replied:
Это и есть SimpleMob

larentoun replied:
А, точно, я то думала это уже для макак

larentoun commented on code/modules/research/xenobiology/xenobiology.dm:
```diff
@@ -183,45 +183,83 @@
-		..()
+	//обработка низших форм: Обезьяны, стока, фарвы, неары, вульпина
+	if (ismonkeybasic(M))
+		var/mob/living/carbon/human/monkey/LF = M
```
LesserForm? LowFricks?

PhantornRU replied:
LesserForm

larentoun commented on code/modules/research/xenobiology/xenobiology.dm:
```diff
@@ -183,45 +183,83 @@
+		if(!src)
+			return
+
+		if(candidates.len)
```
length()

PhantornRU replied:
Это проверка на "не нулевость". len для этого и нужна, она возвращает значение FALSE при NULL.
length() в данном случае будет делать тоже самое, только больше буковок будет написано. Оба эти значения возвращают нули, если списки будут пусты. Если нет, то продолжают работу.
поэтому шо так, шо так. Но в нашем случае в ДМе принято использоваться именно что .len
Да и мне так удобнее сразу у объекты вызывать нужные внутренние функции, а не притягивать сторонние функции.

PhantornRU commented:
@Bizzonium Авоут был пройден.
https://discord.com/channels/617003227182792704/755125334097133628/1014916202784620694

PhantornRU committed:
Ботинки с 50% шансом слетают при подскальзывании, АВОУТ ПРОЙДЕН

Для  ПРа одежды для обезьян
https://discord.com/channels/617003227182792704/755125334097133628/1015906731173351465

PhantornRU committed:
Другие специи могут потерять обувь при подскальзывании

PhantornRU committed:
рефактор низших форм, низшие формы теперь правильно "горят" и не имеют фантомных ног

PhantornRU committed:
Фикс генетик-аур lesser форм

PhantornRU committed:
Исправление неправильного типа, внесение мартыхи в sentience event, т.к. она теперь почти симпл моб

PhantornRU committed:
Исправление дебаг-метода для теста ботинок, выдача возможности "обзывать" питомца от зелья разума.

PhantornRU committed:
Слаймы при использовании зелья разума - получают постоянное имя.

larentoun commented on code/modules/mob/living/simple_animal/slime/slime.dm:
```diff
@@ -1,5 +1,6 @@
 /mob/living/simple_animal/slime
 	name = "grey baby slime (123)"
+	var/rename = FALSE //если слайм был заранее переименован
```
лучше как глагол, "renamed"

PhantornRU replied:
точно-точно

PhantornRU replied:
Переименовал

PhantornRU committed:
Переименование логической переменной

larentoun commented on code/modules/mob/living/carbon/human/species/monkey.dm:
```diff
@@ -8,11 +8,12 @@
 	species_traits = list(NO_EXAMINE)
 	skinned_type = /obj/item/stack/sheet/animalhide/monkey
 	greater_form = /datum/species/human
+	//no_equip = null
```
Аа? Зачем этот коммент?

larentoun commented on code/modules/research/xenobiology/xenobiology.dm:
```diff
@@ -183,45 +183,101 @@
+		if(!src)
+			return
+
+		if(candidates.len)
```
Тут вполне вероятна возможность рантайма в виде `null.len`, замени на `length()`

larentoun replied:
![image](https://user-images.githubusercontent.com/31931237/191931012-d8d5ee6c-cc84-4c12-9ff7-6ee5f19dfac0.png)
Если что, `.len` дает рантайм, если ему дают нулл. Поэтому я говорю использовать `length()`
![image](https://user-images.githubusercontent.com/31931237/191931168-497fbe66-4d59-41f4-8db5-7c72cce65c54.png)

PhantornRU replied:
Ты прав, ты прав

larentoun commented on code/modules/research/xenobiology/xenobiology.dm:
```diff
@@ -183,45 +183,101 @@
+		else
+			to_chat(user, "<span class='notice'>[M] looks interested for a moment, but then looks back down. Maybe you should try again later.</span>")
+			being_used = 0
+			..()
```
Точно не нужен `return ..()`?

larentoun commented on code/modules/research/xenobiology/xenobiology.dm:
```diff
@@ -183,45 +183,101 @@
+		if(!src)
+			return
+
+		if(candidates.len)
```
Опасность `null.len`

larentoun commented on code/modules/research/xenobiology/xenobiology.dm:
```diff
@@ -183,45 +183,101 @@
+		else
+			to_chat(user, "<span class='notice'>[M] выглядел заинтересованым и даже потянулся к зелью, но его резко что-то отвлекло. Стоит попробовать снова попозже.</span>")
+			being_used = 0
+			..()
```
Точно не нужен `return ..()`?

PhantornRU replied:
тогда уж . = ..()
Но это оригинальный код. Здесь нужно же просто глухое проигрывание

PhantornRU replied:
Внёс правку на . = ..()

larentoun replied:
> тогда уж . = ..() Но это оригинальный код. Здесь нужно же просто глухое проигрывание

Ну, я не читала, откуда оно берет ноги, просто спрашиваю)

PhantornRU committed:
Правки кода larentoun

PhantornRU commented:
@Dimach а этот еще более древнее

PhantornRU committed:
Merge remote-tracking branch 'upstream/master220' into monki_mind

PhantornRU committed:
Решение конфликта

PhantornRU commented:
@Dimach мерж конфликт решен

PhantornRU committed:
Merge remote-tracking branch 'upstream/master220' into monki_mind

PhantornRU committed:
Решение конфликта

PhantornRU commented:
@Dimach конфликт вновь решен

PhantornRU commented:
Интересный факт, прошел целый месяц ожидания мержа разума для манки.

larentoun commented:
Интересный факт, снова конфликты

PhantornRU commented:
> Интересный факт, снова конфликты

:(

PhantornRU committed:
Merge remote-tracking branch 'upstream/master220' into monki_mind

PhantornRU committed:
Решение конфликта

Задолбали карты с обезьянками туда-сюда дергать
